cdc/owner/changefeed.go

```diff
@@ -303,15 +315,22 @@ LOOP:
 		return errors.Trace(err)
 	}
 
-	c.initialized = true
+	atomic.StoreInt32(&c.runningStatus, changeFeedRunning)
+	log.Info("changefeed runningStatus",
+		zap.String("changefeed", c.state.ID),
+		zap.Int32("runningStatus", c.runningStatus))
 	return nil
 }
 
 func (c *changefeed) releaseResources(ctx cdcContext.Context) {
-	if !c.initialized {
+	if atomic.LoadInt32(&c.runningStatus) == changeFeedClosed {
 		c.redoManagerCleanup(ctx)
 		return
 	}
+	if atomic.LoadInt32(&c.runningStatus) == changeFeedClosing {
```

Comment:
`c.redoManagerCleanup(ctx)` also needs to be called when changefeed is in `changeFeedClosing` state.